roles/flent/tasks/main.yml

```diff
@@ -0,0 +1,249 @@
+      - type = flent-bench-server-{{ trunc_uuid }}
+  register: serviceip
+  when: workload_args.serviceip is defined and workload_args.serviceip
+
+- block:
+
+  - name: Create service for server pods
+    k8s:
+      definition: "{{ lookup('template', 'service.yml.j2') | from_yaml }}"
+    with_sequence: start=0 count={{ workload_args.pair | default('1')|int }}
+    when: workload_args.serviceip is defined and workload_args.serviceip
+
+  - name: Start Server(s)
+    k8s:
+      definition: "{{ lookup('template', 'server.yml.j2') | from_yaml }}"
+    register: servers
+    with_sequence: start=0 count={{ workload_args.pair | default('1')|int }}
+
+  - name: Wait for pods to be running....
+    k8s_info:
+      kind: Pod
+      api_version: v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - type = flent-bench-server-{{ trunc_uuid }}
+    register: server_pods
+
+  - name: Update resource state to starting servers
+    operator_sdk.util.k8s_status:
+      api_version: ripsaw.cloudbulldozer.io/v1alpha1
+      kind: Benchmark
+      name: "{{ meta.name }}"
+      namespace: "{{ operator_namespace }}"
+      status:
+        state: "Starting Servers"
+
+  when: resource_state.resources[0].status.state == "Building"
+
+- block:
+
+  - name: Get server pods
+    k8s_info:
+      kind: Pod
+      api_version: v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - type = flent-bench-server-{{ trunc_uuid }}
+    register: server_pods
+
+  - name: Update resource state to starting clients
+    operator_sdk.util.k8s_status:
+      api_version: ripsaw.cloudbulldozer.io/v1alpha1
+      kind: Benchmark
+      name: "{{ meta.name }}"
+      namespace: "{{ operator_namespace }}"
+      status:
+        state: "Starting Clients"
+    when: "workload_args.pair|default('1')|int == server_pods | json_query('resources[].status[]')|selectattr('phase','match','Running')|list|length"
+
+
+  when: resource_state.resources[0].status.state == "Starting Servers"
+
+- block:
+
+  - name: Get pod info
+    k8s_info:
+      kind: Pod
+      api_version: v1
+      namespace: '{{ operator_namespace }}'
+      label_selectors:
+        - type = flent-bench-server-{{ trunc_uuid }}
+    register: server_pods
+
+  - block:
+    - name: Start Client(s)
+      k8s:
+        definition: "{{ lookup('template', 'workload.yml.j2') | from_yaml }}"
+      with_items: "{{ server_pods.resources }}"
+      when: workload_args.serviceip is defined and not workload_args.serviceip|default('false') and server_pods.resources|length > 0
+
+    - name: Start Client(s) - ServiceIP
+      k8s:
+        definition: "{{ lookup('template', 'workload.yml.j2') | from_yaml }}"
+      with_items: "{{ serviceip.resources }}"
+      when: workload_args.serviceip is defined and workload_args.serviceip and serviceip.resources|length > 0
+
+    - name: Validate input
+      fail:
+        msg: 'ServiceIP setting not defined'
+      ignore_errors: yes
```

Comment:
Can you explain what the goal of this part is? I'm a little lost on why theres a fail with an ignore_errors. Thanks!

Reply:
It was for just input validation, but it's not needed so I'll remove it.